cmd/testworkflow-init/data/state.go

```diff
@@ -54,37 +49,54 @@ func (s *state) SetOutput(ref, name string, value interface{}) {
 	if err == nil {
 		s.Output[name] = string(v)
 	} else {
-		fmt.Printf("Warning: couldn't save '%s' (%s) output: %s\n", name, ref, err.Error())
+		output.Std.Warnf("warn: couldn't save '%s' (%s) output: %s\n", name, ref, err.Error())
 	}
 }
 
-func (s *state) GetSelfStatus() string {
-	if Step.Executed {
-		return string(Step.Status)
+func (s *state) GetStep(ref string) *StepData {
+	if s.Steps[ref] == nil {
+		s.Steps[ref] = &StepData{Ref: ref}
 	}
-	v := s.GetStep(Step.Ref)
-	if v.Status != StepStatusPassed {
-		return string(v.Status)
+	if s.Steps[ref].Condition == "" {
+		s.Steps[ref].Condition = "passed"
```

Comment:
should it be const?

Reply:
It rather shouldn't - `passed` is just a built-in variable, and it could be any kind of expression